File: py/Versioning/GitVersioning.py

import os

from .Versioning import Versioning

class GitVersioning(Versioning):
    def __init__(self, projectRoot, metaFolder):
        self._projectRoot = projectRoot
        self._metaFolder = metaFolder

    def name(self):
        return "Git"
    
    def projectRoot(self):
        return self._projectRoot
    
    def metaFolder(self): # e.g.: the .git folder
        return self._metaFolder
    
    def status(self):
        return ""
        
    def openUI(self):
        os.system(f"git --git-dir=" + self._metaFolder + " --work-tree=" + self._projectRoot + " gui")
        

File: java/src/main/java/de/addiks/einsicht/versioning/GitVersioning.java
package de.addiks.einsicht.versioning;

import java.nio.file.Path;

public class GitVersioning implements Versioning {
    private final Path projectRoot;
    private final Path metaFolder;

    public GitVersioning(Path projectRoot, Path metaFolder) {
        this.projectRoot = projectRoot;
        this.metaFolder = metaFolder;
    }

    @Override
    public String name() {
        return "Git";
    }

    @Override
    public Path projectRoot() {
        return projectRoot;
    }

    @Override
    public Path metaFolder() {
        return metaFolder;
    }

    @Override
    public String status() {
        return ""; // TODO
    }

    @Override
    public void openUI() {
        // "git --git-dir=" + metaFolder + " --work-tree=" + projectRoot + " gui"
    }
}
